src/services/quotes/quote-sources/barter-quote-source.ts

```diff
@@ -1,11 +1,10 @@
 import { Chains } from '@chains';
-import { Address, Chain, TimeString } from '@types';
+import { Address, Chain, ChainId, TimeString } from '@types';
 import { IFetchService } from '@services/fetch';
 import { calculateDeadline, isSameAddress, substractPercentage } from '@shared/utils';
 import { Addresses } from '@shared/constants';
-import { QuoteParams, QuoteSourceMetadata, SourceQuoteResponse } from './types';
+import { IQuoteSource, QuoteParams, QuoteSourceMetadata, SourceQuoteResponse } from './types';
 import { calculateAllowanceTarget, checksum, failed } from './utils';
-import { AlwaysValidConfigAndContextSource } from './base/always-valid-source';
 
```

Comment:
Let's do something a little bit different so that it's easier to configure
1. Let's have a constant `Record<ChainId, string>`, where the value would be `eth` or `arb` for now
2. We can then calculate the chains on the metadata based on the keys of this record
3. As part of the config, let's take two properties: `customSubdomain` and `authHeader`, both required
4. With these two properties, and the constant, we can build the request properly